enterprise/backup/src/test/java/org/neo4j/backup/BackupFlowTest.java

```diff
@@ -0,0 +1,267 @@
+        expectedException.expect( exceptionContainsSuppressedThrowable( firstCause ) );
+        expectedException.expect( exceptionContainsSuppressedThrowable( secondCause ) );
+        expectedException.expect( CommandFailed.class );
+        expectedException.expectMessage( "Failed to run a backup using the available strategies. Perhaps using the wrong port?" );
+
+        // when
+        subject.performBackup( onlineBackupContext );
+    }
+
+    @Test
+    public void errorsDuringConsistencyCheckAreWrappedAsCommandFailed() throws CommandFailed, IOException, ConsistencyCheckIncompleteException
+    {
+        // given
+        anyStrategyPasses();
+        when( requiredArguments.isDoConsistencyCheck() ).thenReturn( true );
+        when( consistencyCheckService.runFullConsistencyCheck( any(), any(),
+                eq(progressMonitorFactory), any( LogProvider.class ), any( FileSystemAbstraction.class),
+                eq(false), any(File.class), any( CheckConsistencyConfig.class ) ) )
+                .thenThrow( new IOException( "Predictable message" ) );
+
+        // then
+        expectedException.expect( CommandFailed.class );
+        expectedException.expectMessage( containsString( "Failed to do consistency check on backup: Predictable message" ) );
+
+        // when
+        subject.performBackup( onlineBackupContext );
+    }
+
+    @Test
+    public void commandFailedWhenConsistencyCheckFails() throws IOException, ConsistencyCheckIncompleteException, CommandFailed
+    {
+        // given
+        anyStrategyPasses();
+        when( requiredArguments.isDoConsistencyCheck() ).thenReturn( true );
+        when( consistencyCheckResult.isSuccessful() ).thenReturn( false );
+        when( consistencyCheckService.runFullConsistencyCheck( any(), any(),
+                eq(progressMonitorFactory), any( LogProvider.class ), any( FileSystemAbstraction.class),
+                eq(false), any(File.class), any( CheckConsistencyConfig.class ) ) )
+                .thenReturn( consistencyCheckResult );
+
+        // then
+        expectedException.expect( CommandFailed.class );
+        expectedException.expectMessage( "Inconsistencies found" );
+
+        // when
+        subject.performBackup( onlineBackupContext );
+    }
+
+    @Test
+    public void havingNoStrategiesCausesAllSolutionsFailedException() throws CommandFailed
+    {
+        // given there are no strategies in the solution
+        subject = new BackupFlow( consistencyCheckService, outsideWorld, logProvider, progressMonitorFactory, Collections.emptyList() );
+
+        // then we want a predictable exception (instead of NullPointer)
+        expectedException.expect( CommandFailed.class );
+        expectedException.expectMessage( "Failed to run a backup using the available strategies. Perhaps using the wrong port?" );
+
+        // when
+        subject.performBackup( onlineBackupContext );
+    }
+
+    private void anyStrategyPasses()
```

Comment:
Add annotation or delete test?

Reply:
As it turns out, this is used in the above tests to simplify configuring strategy mocks.
I have added a minor java-doc to this method.